python/ray/train/_internal/backend_executor.py

```diff
@@ -663,14 +663,20 @@ def end_training():
         results = self.get_with_failure_handling(futures)
         return results
 
-    def report_final_run_status(self, errored=False):
-        """Report the final train run status and end time to TrainStateActor."""
+    def report_final_run_status(
+        self, errored=False, failed_rank=None, stack_trace=None
+    ):
+        """Report the final train run status, error, and end time to TrainStateActor."""
         if self.state_tracking_enabled:
-            from ray.train._internal.state.schema import RunStatusEnum
+            from ray.train._internal.state.schema import (
+                MAX_ERROR_STACK_TRACE_LENGTH,
+                RunStatusEnum,
+            )
 
             if errored:
                 run_status = RunStatusEnum.ERRORED
-                status_detail = "Terminated due to an error in the training function."
+                status_detail = f"Rank {failed_rank} worker raised an error: \n"
+                status_detail += stack_trace[-MAX_ERROR_STACK_TRACE_LENGTH:]
```

Comment:
Add a comment here explaining this logic? (Just make it obvious that it's only showing the end)